### PR TITLE
[MI-2418] Add test cases for kv.go, util.go and encode_key.go

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

File: server/plugin/kv_test.go [hunk 1]
```diff
@@ -145,6 +147,96 @@ func TestDeleteUser(t *testing.T) {
...
+		},
+		{
+			description: "GetAllUsers: KVList gives error",
+			setupTest:   func() {},
+			setupAPI: func(a *plugintest.API) {
+				a.On("KVList", testutils.GetMockArgumentsWithType("int", 2)...).Return(
+					nil, testutils.GetInternalServerAppError("error in loading the KVList"),
+				)
+			},
+			expectedError: ": , error in loading the KVList",
```
Comment (raghavaggarwal2308) [inline server/plugin/kv_test.go:184]: Why does the error start with `: ,`

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/kv_test.go (hunk 1 above)
Comment (ayusht2810) [reply server/plugin/kv_test.go:184]: it gets added by the function itself

File: server/plugin/util_test.go [hunk 2]
```diff
@@ -310,3 +310,143 @@ func TestHandleClientError(t *testing.T) {
...
+			errorMessage: constants.ErrorChannelPermissionsForUser,
+		},
+		{
+			description: "HasChannelPermissions: channel is a direct channel",
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetChannel", mock.AnythingOfType("string")).Return(
+					testutils.GetChannel(model.CHANNEL_DIRECT), nil,
+				)
+			},
+			statusCode:   http.StatusBadRequest,
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/plugin/util_test.go:417]: yes

File: server/plugin/util_test.go [hunk 3]
```diff
@@ -310,3 +310,143 @@ func TestHandleClientError(t *testing.T) {
...
+				api.On("LogDebug", testutils.GetMockArgumentsWithType("string", 3)...).Return()
+			},
+			statusCode:   http.StatusInternalServerError,
+			errorMessage: constants.ErrorInsufficientPermissions,
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			assert := assert.New(t)
+
+			p, api := setupTestPlugin(&plugintest.API{}, nil)
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/plugin/util_test.go:438]: we are getting errors in other test cases on doing that

File: server/plugin/kv_test.go [hunk 4]
```diff
@@ -145,6 +148,96 @@ func TestDeleteUser(t *testing.T) {
...
+					}, nil,
+				)
+			},
+		},
+		{
+			description: "GetAllUsers: KVList gives error",
+			setupTest:   func() {},
+			setupAPI: func(a *plugintest.API) {
+				a.On("KVList", testutils.GetMockArgumentsWithType("int", 2)...).Return(
+					nil, testutils.GetInternalServerAppError("error in loading the KVList"),
```
Comment (Kshitij-Katiyar) [inline server/plugin/kv_test.go:182]: KVList is a function, not an actual list. So I think this should be better please change everywhere
```suggestion
					nil, testutils.GetInternalServerAppError("error in loading the user list"),
```

**Review by Kshitij-Katiyar — APPROVED** [review]

**Pull request merged**